frontend/src/components/recipepage.jsx

```diff
@@ -0,0 +1,100 @@
+import './recipepage.css';
+import React, {useState, useEffect} from "react";
+
+  
+export default function Recipepage({name,  desc, image, ingredients,instructions}){
+  const [newIngredient, setNewIngredient] = React.useState('');
+  const [newInstruction, setNewInstruction] = React.useState('');
+  
+
+const addIngredient = () => {
+    var ul = document.getElementById("ingredientList");
+    var li = document.createElement("li");
+    li.appendChild(document.createTextNode(newIngredient));
+    ul.appendChild(li);
+  }
+  
+  const addInstruction = () => {
+    var ol = document.getElementById("instructionList");
+    var li = document.createElement("li");
+    li.appendChild(document.createTextNode(newInstruction));
+    ol.appendChild(li);
+  }
+
+  useEffect(() => {
+    if(addIngredient === '') return;
+
+    const requestOptions = {
```

Comment:
IT doesn't look like your requestOptions are ever being called, and thus your request isn't quite working. 

Here is an example of a working PUT request (however, it looks like you are attempting to use a POST request, so you might have to change PUT -> POST to get it to work as intended)

```  let addIngredientToDB = (newIngredient) => {
    let Obj = {newIngredient: newIngredient}
    fetch(`http://localhost:3001/api/recipe/${link}/ingredient`, {
        headers : { 
          'Content-Type': 'application/json',
          'Accept': 'application/json'
        },
        method: 'PUT',
        body: JSON.stringify(Obj)
    }).then(response => response.json())
  }```